x-pack/plugins/lens/public/types.ts

```diff
@@ -345,12 +344,21 @@ export type VisualizationDimensionEditorProps<T = unknown> = VisualizationConfig
   setState: (newState: T) => void;
 };
 
+export type AccessorConfig =
+  | string
```

Comment:
I wonder if it wouldn't be better to not support this type anymore and convert all the accessors to the format below (`{columnId:... } `). We already introduce 6 checks in layer_panel only to check if `typeof accessorConfig !== 'string'`...